### PR TITLE
Mention keyboard shortcuts in tooltips (#1941)

### PR DESCRIPTION
**Pull request by mejo-**
Doesn't show the shortcuts on mobile (they don't apply there).

recording

* Target version: master

### CODE REVIEW

**Comment by azul** [comment]
I'm unsure about this one. I see github is doing the same and i see the point of making keyboard shortcuts discoverable. At the same time it's always there for everyone and clutters the ui a bit. I try to use the markdown shortcuts such as `_italic_` instead of ctrl+i myself.

File: src/mixins/menubar.js [hunk 1]
```diff
@@ -115,6 +127,7 @@ export default [
 	},
 	{
 		label: t('text', 'Unordered list'),
+		keyboard: t('text', 'Shift') + '+' + t('text', 'Ctrl') + '+8',
```
Comment (azul) [inline src/mixins/menubar.js:130]: I feel like constructing these strings could be moved to a helper function. Something like:
```
keyboardShortcut('8', shift: true, ctrl: true)
```

**Review by azul — APPROVED** [review]
Codewise looks good to me. I have one small comment I will add in line.

**Comment by mejo-** [comment]
> I'm unsure about this one. I see github is doing the same and i see the point of making keyboard shortcuts discoverable. At the same time it's always there for everyone and clutters the ui a bit. I try to use the markdown shortcuts such as `_italic_` instead of ctrl+i myself.

I started looking into this after I got the feedback that people didn't know at all about formatting shortcuts in Nextcloud Text. That's why I think it's a good thing to make it obvious.

As discussed in #1941, the idea is to mention the keyboard shortcuts in toolbar tooltips in a first step and add a motion with some explanatory text and details about the markdown shortcuts there.

I took a look at several web editors (Google Docs, Github, Gitlab, Discourse, Etherpad) and they all mention keyboard shortcuts in their tooltips :blush:

**Review by jancborchardt — CHANGES_REQUESTED** [review]
The tooltips are good as discussed, since they only appear after some time. :)

But in the headings action menu and also for blockquote, having the shortcut directly in the menu is indeed a bit cluttered as @azul mentions. Is it possible to put the shortcut info in a tooltip there too?

**Comment by mejo-** [comment]
> But in the headings action menu and also for blockquote, having the shortcut directly in the menu is indeed a bit cluttered as @azul mentions. Is it possible to put the shortcut info in a tooltip there too?

Unfortunately neither `ActionButton` nor `PopoverMenuItem` support setting the `title` property for `button` element so far. I opened a PR in nextcloud-vue to fix that: https://github.com/nextcloud/nextcloud-vue/pull/2375. With this PR being merged (and released), it's possible to set tooltips for actions in submenus:

recording

**Comment by mejo-** [comment]
After some further discussion and brainstorming with @skjnldsv, I went with using `v-tooltip` instead, which is the prefered way for tooltips in Nextcloud anyways.

Also, @skjnldsv made the valid point that tooltips in the headings submenu would be confusing (basically tooltip inside a tooltip). Therefore I went with `Heading (Ctrl+Shift+1…6)` as a generic headings tooltip and no tooltips for the concrete levels. What do you think about it?

See the updated gif above.

File: src/mixins/menubar.js (hunk 1 above)
Comment (mejo-) [reply src/mixins/menubar.js]: Did this now, thanks to pair programming with @azul  :)

**Review by nimishavijay — APPROVED** [review]
Looks good design wise! Agreed with using `Ctrl+Shift+1...6` because otherwise it would be repeated many times in the submenu

**Comment by mejo-** [comment]
@jancborchardt could you take a quick look? Github tells me that you requested changes and you probably have to approve before Github lets us merge :)

**Pull request merged**